ipatests/test_integration/test_ntp_options.py

```diff
@@ -0,0 +1,196 @@
+        server_install = tasks.install_master(
+            self.master, setup_dns=False,
+            extra_args=['--ntp-pool=%s' % ntp_pool,
+                        '--ntp-server=%s' % ntp_server])
+        assert expected_msg in server_install.stderr_text
+        cmd = self.master.run_command(['cat', paths.CHRONY_CONF])
+        assert ntp_pool in cmd.stdout_text
+        assert ntp_server in cmd.stdout_text
+        replica_install = self.install_replica('--ntp-pool=%s' % ntp_pool,
+                                               '--ntp-server=%s' % ntp_server)
+        assert expected_msg in replica_install.stderr_text
+        cmd = self.replica.run_command(['cat', paths.CHRONY_CONF])
+        assert ntp_pool in cmd.stdout_text
+        assert ntp_server in cmd.stdout_text
+        client_install = self.install_client('--ntp-pool=%s' % ntp_pool,
+                                             '--ntp-server=%s' % ntp_server)
+        assert expected_msg in client_install.stderr_text
+        cmd = self.client.run_command(['cat', paths.CHRONY_CONF])
+        assert ntp_pool in cmd.stdout_text
+        assert ntp_server in cmd.stdout_text
+        tasks.uninstall_master(self.replica)
+        self.cleanup()
+
+    def test_server_and_client_install_with_mixed_options(self):
+        """
+        test to verify ipa-server and ipa-client install with
+         --ntp-server and -N options
+        """
+        ntp_server = "1.pool.ntp.org"
+        exp_str = ("error: --ntp-server cannot be used"
+                   " together with --no-ntp")
+        exp_pool_str = ("error: --ntp-pool cannot be used"
+                        " together with --no-ntp")
+        args1 = ['ipa-server-install', '-N', '--ntp-server=%s' % ntp_server]
+        server_install = self.master.run_command(args1, raiseonerr=False)
+        assert server_install.returncode == 2
+        assert exp_str in server_install.stderr_text
+        args2 = ['ipa-client-install', '-N', '--ntp-server=%s' % ntp_server]
+        client_install = self.client.run_command(args2, raiseonerr=False)
+        assert client_install.returncode == 2
+        assert exp_str in client_install.stderr_text
+        args3 = ['ipa-client-install', '-N',
+                 '--ntp-pool=%s' % ntp_server.lstrip('1.')]
+        client_install = self.client.run_command(args3, raiseonerr=False)
+        assert client_install.returncode == 2
+        assert exp_pool_str in client_install.stderr_text
+
+    def test_two_step_replica_install_using_ntp_options(self):
+        """
+        test to verify 2-step ipa-replica install with --ntp-server, --ntp-pool
+        and -N options
+        """
+        ntp_server = "1.pool.ntp.org"
+        ntp_pool = "pool.ntp.org"
+        exp_str = "NTP configuration cannot be updated during promotion"
+        tasks.install_master(self.master, setup_dns=False)
+        tasks.install_client(self.master, self.replica)
+        replica_install = self.replica.run_command(
```

Comment:
What if these methods would look like this ? :
```python
    def test_two_step_replica_install_using_ntp_options(self):
        """
        test to verify that replica promotion with --ntp-server,
        --ntp-pool or --no-ntp / -N option would fail
        """
        ntp_server = "1.pool.ntp.org"
        ntp_pool = "pool.ntp.org"
        exp_str = "NTP configuration cannot be updated during promotion"
       
        tasks.install_master(self.master, setup_dns=False)
        tasks.install_client(self.master, self.replica)
        
        try:
            replica_install = self.replica.run_command(
                ['ipa-replica-install', '--no-ntp'], 
                raiseonerr=False)
            assert replica_install.returncode == 1
            assert exp_str in replica_install.stderr_text

            replica_install = self.replica.run_command(
                ['ipa-replica-install', '--ntp-server=%s' % ntp_server],
                raiseonerr=False)
            assert replica_install.returncode == 1
            assert exp_str in replica_install.stderr_text

            replica_install = self.replica.run_command(
                ['ipa-replica-install', '--ntp-pool=%s' % ntp_pool],
                raiseonerr=False)
            assert replica_install.returncode == 1
            assert exp_str in replica_install.stderr_text

        finally:  # to cleanup always even when test fails
            self.cleanup()
```

Reply:
looks good